### PR TITLE
LPS-153141 Logic for Undo/Redo

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

File: modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/style-book-editor/reducer.js [hunk 1]
```diff
@@ -50,14 +52,69 @@ export default function reducer(state, action) {
...
+			return {
+				...state,
+				redoHistory: [
+					...redoHistory,
+					{
+						name,
+						value: {
+							...value,
+							name: frontendTokensValues[name]?.name,
+							value: frontendTokensValues[name].value,
```
Comment (victorg1991) [inline modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/style-book-editor/reducer.js:92]: This is crashing when clicking remove button in a color token. Can we take the previous value outside here?

Something like

```javascript
const {name, value} = action;
const {frontendTokensValues, redoHistory, undoHistory} = state;

const previousValue = frontendTokensValues[name];

return {
... state,
redoHistory: [

{name, previousValue} ]

```

**Comment by victorg1991** [comment]
ci:forward

**Pull request closed without merging**